eng/scripts/build.ps1

```diff
@@ -9,75 +9,96 @@ param(
     [switch]$format,
     [switch]$tidy,
     [string]$config = "autorest.md",
-    [string]$goExtension = "@autorest/go@4.0.0-preview.27",
+    [string]$goExtension = "@autorest/go@4.0.0-preview.28",
     [string]$outputFolder
 )
 
 . (Join-Path $PSScriptRoot .. common scripts common.ps1)
 
-function Process-Sdk () {
+function Process-Sdk ()
+{
     $currentDirectory = Get-Location
-    if ($clean) {
+    if ($clean)
+    {
         Write-Host "##[command]Executing go clean -v ./... in " $currentDirectory
         go clean -v ./...
         if ($LASTEXITCODE) { exit $LASTEXITCODE }
     }
 
-    if ($cleanGenerated) {
+    if ($cleanGenerated)
+    {
         Write-Host "##[command]Cleaning auto-generated files in" $currentDirectory
         Remove-Item "zz_generated_*"
     }
 
-    if ($generate) {
+    if ($generate)
+    {
         Write-Host "##[command]Executing autorest.go in " $currentDirectory
         $autorestPath = "./" + $config
-
-        if ($outputFolder -eq '') {
+        
+        if ($outputFolder -eq '')
+        {
             $outputFolder = $currentDirectory
         }
         autorest --use=$goExtension --go --track2 --output-folder=$outputFolder --file-prefix="zz_generated_" --clear-output-folder=false $autorestPath
-        if ($LASTEXITCODE) {
+        if ($LASTEXITCODE)
+        {
             Write-Host "##[error]Error running autorest.go"
             exit $LASTEXITCODE
         }
     }
 
-    if ($format) {
+    if ($format)
+    {
         Write-Host "##[command]Executing gofmt -s -w . in " $currentDirectory
         gofmt -s -w .
-        if ($LASTEXITCODE) {exit $LASTEXITCODE}
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
     }
 
-    if ($tidy) {
+    if ($tidy)
+    {
         Write-Host "##[command]Executing go mod tidy in " $currentDirectory
         go mod tidy
-        if ($LASTEXITCODE) {exit $LASTEXITCODE}
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
     }
 
-    if (!$skipBuild) {
+    if (!$skipBuild)
+    {
         Write-Host "##[command]Executing go build -x -v ./... in " $currentDirectory
         go build -x -v ./...
         Write-Host "##[command]Build Complete!"
-        if ($LASTEXITCODE) {exit $LASTEXITCODE}
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
     }
 
-    if ($vet) {
+    if ($vet)
+    {
         Write-Host "##[command]Executing go vet ./... in " $currentDirectory
         go vet ./...
     }
 }
 
-try {
+try
+{
     $startingDirectory = Get-Location
 
     $sdks = Get-AllPackageInfoFromRepo $filter
 
-    foreach ($sdk in $sdks) {
+    foreach ($sdk in $sdks)
+    {
         Push-Location $sdk.DirectoryPath
         Process-Sdk
         Pop-Location
     }
+
```

Comment:
future clean-up I would capture `Join-Path $RepoRoot "sdk" $filter` once instead of duplicating it 3 times.